lib/src/Uncertainty/Algorithm/Simulation/SubsetSampling.cxx

```diff
@@ -0,0 +1,572 @@
+{
+  const UnsignedInteger N = currentPointSample_.getSize();
+  const UnsignedInteger Nc = std::max<UnsignedInteger>(1, targetProbability_ * N);
+  Matrix IndicatorMatrice(Nc, N / Nc);
+  NumericalPoint correlationSequence(N / Nc - 1);
+  NumericalScalar currentFailureProbability2 = pow(currentFailureProbability, 2.);
+  for (UnsignedInteger i = 0; i < N / Nc; ++ i)
+  {
+    for (UnsignedInteger j = 0; j < Nc; ++ j)
+    {
+      IndicatorMatrice(j, i) = getEvent().getOperator()(currentLevelSample_[ i*Nc+j ][0], threshold);
+    }
+  }
+  for (UnsignedInteger k = 0; k < N / Nc - 1; ++ k)
+  {
+    for (UnsignedInteger j = 0; j < Nc; ++ j)
+    {
+      for (UnsignedInteger l = 0; l < N / Nc - k - 1; ++ l)
+      {
+        correlationSequence[k] += 1.0 * IndicatorMatrice(j, l) * IndicatorMatrice(j, l + (k + 1));
+      }
+    }
+    correlationSequence[k] /= 1.0 * N - 1.0 * (k + 1) * Nc;
+    correlationSequence[k] -= currentFailureProbability2;
+  }
+  const NumericalScalar R0 = currentFailureProbability * (1.0 - currentFailureProbability);
+  NumericalPoint rho = ((1.0 / R0) * correlationSequence);
+  NumericalScalar gamma = 0.0;
+  for (UnsignedInteger k = 0; k < N / Nc - 1; ++ k)
+  {
+    gamma += 2.0 * (1.0 - (k + 1) * 1.0 * Nc / N) * rho[k];
+  }
+  return gamma;
+}
+
+
+/* Iterate one step of the algorithm */
+void SubsetSampling::generatePoints(NumericalScalar threshold)
+{
+  UnsignedInteger maximumOuterSampling = getMaximumOuterSampling();
+  UnsignedInteger blockSize = getBlockSize();
+  Distribution randomWalk(ComposedDistribution(ComposedDistribution::DistributionCollection(dimension_, Uniform(-0.5*proposalRange_, 0.5*proposalRange_))));
+  UnsignedInteger N = currentPointSample_.getSize(); // total sample size
+  UnsignedInteger Nc = targetProbability_ * N; //number of seeds (also = maximumOuterSampling*blockSize)
+
+  for (UnsignedInteger i = 0; i < maximumOuterSampling; ++ i)
+  {
+    NumericalSample inputSample(blockSize, dimension_);
+    for (UnsignedInteger j = 0; j < blockSize; ++ j)
+    {
+      // assign the new point to the seed, seed points being regrouped at the beginning of the sample
+      if (i*blockSize+j >= Nc)
+      {
+        currentPointSample_[i * blockSize + j] = currentPointSample_[ i*blockSize+j-Nc ];
+        currentLevelSample_[i * blockSize + j] = currentLevelSample_[ i*blockSize+j-Nc ];     
+      }
+
+      // generate a new point
+      NumericalPoint oldPoint(currentPointSample_[i * blockSize + j]);
+      NumericalPoint newPoint(oldPoint + randomWalk.getRealization());
+
+      // 1. accept / reject new components
+      NumericalPoint uniform(RandomGenerator::Generate(dimension_));
+      for (UnsignedInteger k = 0; k < dimension_; ++ k)
+      {
+        // compute ratio
+        NumericalScalar ratio = std::min(1.0, exp(0.5 * (oldPoint[k] * oldPoint[k] - newPoint[k] * newPoint[k])));
```

Comment:
No need to take the minimum as the next test will reject the proposal as soon as the ratio is not less than 1